### PR TITLE
Add assertion for always save nemo add model parallel size

### PR DESCRIPTION
**Pull request by BoxiangW**
# What does this PR do ?

`always_save_nemo` option is not supported yet to work with model parallel. Added assertions to avoid future confusions.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/utils/callbacks/nemo_model_checkpoint.py [hunk 1]
```diff
@@ -208,6 +208,8 @@ def on_save_checkpoint(self, trainer, pl_module, checkpoint):
         app_state = AppState()
         if app_state.model_parallel_size is not None and app_state.model_parallel_size > 1:
             logging.warning(f'always_save_nemo will slow down training for model_parallel > 1.')
+        if app_state.context_parallel_size is not None and app_state.context_parallel_size > 1:
+            logging.warning(f'always_save_nemo will slow down training for context_parallel > 1.')
```
Comment (maanug-nv) [inline nemo/utils/callbacks/nemo_model_checkpoint.py:212]: trying to understand this, why will it slow down training? does it take longer to checkpoint with both cp>1 and `always_save_nemo`? is it still a slowdown if we are using async checkpointing?

**Comment by maanug-nv** [comment]
Seems like we don't save '.nemo' in 2.0 right?

File: nemo/utils/callbacks/nemo_model_checkpoint.py (hunk 1 above)
Comment (BoxiangW) [reply nemo/utils/callbacks/nemo_model_checkpoint.py:212]: It is actually `always_save_nemo` is not supported with any model parallel including CP

**Comment by BoxiangW** [comment]
Yes, this was an NVbug about NeMo 1.0. We are not going to save .nemo in 2.0 right now

File: nemo/utils/callbacks/nemo_model_checkpoint.py (hunk 1 above)
Comment (maanug-nv) [reply nemo/utils/callbacks/nemo_model_checkpoint.py:212]: Ok, then 'slowdown' is misleading. But then I think the 'error' in exp_manager is sufficient, do we need this warning as well?

Comment (BoxiangW) [reply nemo/utils/callbacks/nemo_model_checkpoint.py:212]: I think we can remove this, only use the exp_manager one then.

**Comment by maanug-nv** [comment]
One last request, can we add a test in `test_exp_manager.py` to ensure this configuration will fail?

**Review by maanug-nv — DISMISSED** [review]
Approving. good to merge if tests pass

**Comment by BoxiangW** [comment]
@maanug-nv  Can you help approve this again? It just passed all tests.

**Review by maanug-nv — APPROVED** [review]

**Pull request merged**